homeassistant/components/battery.py

```diff
@@ -0,0 +1,180 @@
+            _LOGGER.error("Missing configuration data for battery %s", bat)
+            continue
+
+        battery_type = bat_config.get('type', DEFAULT_TYPE)
+        attribute = bat_config.get('attribute', DEFAULT_ATTRIBUTE)
+        device_entity_id = bat_config.get('entity_id')
+        unit_of_measurement = bat_config.get('unit_of_measurement',
+                                             DEFAULT_UNIT)
+        charged_level = bat_config.get('charged_level', DEFAULT_CHARGED)
+
+        entity_id = DOMAIN + '.' + bat
+
+        battery = Battery(hass, battery_type, attribute, device_entity_id,
+                          unit_of_measurement, charged_level)
+        battery.entity_id = entity_id
+
+        battery.update_ha_state()
+        battery.check_battery_state_change(None, None, None)
+
+        batteries.append(battery)
+        # main command to monitor battery of device
+        track_state_change(hass, battery.device,
+                           battery.check_battery_state_change)
+
+    if not batteries:
+        _LOGGER.error("No batteries added")
+        return False
+
+    # Tells the bootstrapper that the component was successfully initialized
+    return True
+
+
+class Battery(Entity):  # pylint: disable=too-many-instance-attributes
+    """ Represents a Battery in Home Assistant. """
+    def __init__(self, hass, battery_type, attribute, device_entity_id,
+                 unit_of_measurement, charged_level):
+        # pylint: disable=too-many-arguments
+        self.hass = hass
+        self.device = device_entity_id
+        self._battery_type = battery_type
+        self._attribute = attribute
+        self._batterylevel = 'not set'
+        self._status = 'not set'
+        self._unit_of_measurement = unit_of_measurement
+        self._charged_level = charged_level
+
+    @property
+    def state(self):
+        return self._batterylevel
+
+    @property
+    def unit_of_measurement(self):
+        """ Unit of measurement of this entity """
+        return self._unit_of_measurement
+
+    @property
+    def state_attributes(self):
+        if self._battery_type == 'state':
+            return {
+                ATTR_STATUS: self._status,
+                ATTR_BATTERY_TYPE: self._battery_type
+            }
+        elif self._battery_type == 'attribute':
+            return {
+                ATTR_STATUS: self._status,
+                ATTR_BATTERY_TYPE: self._battery_type,
+                ATTR_ATTRIBUTE: self._attribute
+            }
+
+    def check_battery_state_change(self, entity, old_state, new_state):
+        # pylint: disable=too-many-branches
+        """ Function to perform the battery checking for a device """
+        if self._battery_type != 'attribute' and self._battery_type != 'state':
+            return False
+
+        if new_state is None:
+            new_state = self.hass.states.get(self.device)
+
+        if (new_state is None or
+                ((self._battery_type == 'attribute') and
```

Comment:
`x == y` is evaluated before `and`, `and` is evaluated before `or`, `not` evaluated before `and`, so you don't need the parentheses. You can use the outer parentheses to achieve the line break without using `\`.